actions/generate-schedule/scheduler/scheduler.go

```diff
@@ -0,0 +1,98 @@
+package scheduler
+
+import (
+	"fmt"
+	"github.com/spinnaker/rotation-scheduler/actions/generate-schedule/users"
+	"github.com/spinnaker/rotation-scheduler/proto/schedule"
+	"time"
+)
+
+const (
+	DateFormat = "Mon 02 Jan 2006"
+)
+
+// Scheduler creates or extends a new rotation schedule.
+type Scheduler struct {
+	userSource        users.Source
+	shiftDurationDays int
+}
+
+// NewScheduler creates a new Scheduler. All args are required.
+func NewScheduler(userSource users.Source, shiftDurationDays int) (*Scheduler, error) {
+	if userSource == nil {
+		return nil, fmt.Errorf("no user source specificed")
+	}
+
+	if shiftDurationDays <= 0 {
+		return nil, fmt.Errorf("shift duration invalid. Must be greater than 0, was: %v", shiftDurationDays)
+	}
+
+	return &Scheduler{
+		userSource:        userSource,
+		shiftDurationDays: shiftDurationDays,
+	}, nil
+}
+
+// Schedule creates a new Schedule from start (inclusive) to stop (exclusive). Will return a 1-entry schedule if
+// stop is after start.
```

Comment:
> Will return a 1-entry schedule if stop is after start.

Wait, what does this mean? Shouldn't `stop` always be after `start`?

Reply:
Fixed - doc line was flipped.